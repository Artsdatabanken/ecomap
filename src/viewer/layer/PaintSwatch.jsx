import React from 'react'
import checkboard from './checkerboard.png'

const PaintSwatch = ({color, onClick}) =>
  <div onClick={(e) => { onClick(e) }} style={{
    position: 'absolute',
    top: '14px',
    right: '14px',
    height: '40px',
    width: '40px',
    borderRadius: '50%',
    backgroundImage: 'linear-gradient(to bottom, ' + color + ', ' + color + '), url(' + checkboard + ')'
  }}>&nbsp;</div>

export default PaintSwatch
