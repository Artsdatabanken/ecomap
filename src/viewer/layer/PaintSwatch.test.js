import React from 'react'
import renderer from 'react-test-renderer'
import muiTheme from '../../theme_jest'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import PaintSwatch from './PaintSwatch'

it('loading standard tile', () => {
  const tree = renderer.create(
    <MuiThemeProvider muiTheme={muiTheme}>
      <PaintSwatch color='#ffeedd' />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
