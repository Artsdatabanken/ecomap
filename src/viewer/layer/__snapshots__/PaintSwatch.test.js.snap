// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading standard tile 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "linear-gradient(to bottom, #ffeedd, #ffeedd), url(checkerboard.png)",
      "borderRadius": "50%",
      "height": "40px",
      "position": "absolute",
      "right": "14px",
      "top": "14px",
      "width": "40px",
    }
  }
>
  Â 
</div>
`;
